internal/core/command/controller/messaging/external/external.go

```diff
@@ -124,6 +134,78 @@ func commandQueryHandler(responseTopic string, qos byte, retain bool, dic *di.Co
 	}
 }
 
+func commandRequestHandler(dic *di.Container) mqtt.MessageHandler {
+	return func(client mqtt.Client, message mqtt.Message) {
+		lc := bootstrapContainer.LoggingClientFrom(dic.Get)
+		messageBusInfo := container.ConfigurationFrom(dic.Get).MessageQueue
+		qos := messageBusInfo.External.QoS
+		retain := messageBusInfo.External.Retain
+
+		// expected command request topic scheme: #/<device>/<command-name>/<method>
+		topicLevels := strings.Split(message.Topic(), "/")
+		length := len(topicLevels)
+		deviceName := topicLevels[length-3]
```

Comment:
Should validate length is sufficient. i.e. > 3

Reply:
In this case what topic should I publish the error response ? (since I cannot construct response topic here)
Also I cannot get the requestId for the envelope here, but if we validate the requestEnvelope first there's the same problem: what topic to response to if there's error?

Can we just publish to `edgex/command/response` ?
https://github.com/edgexfoundry/edgex-go/blob/b7de7b775397295cb69718a186a97b7526882c59/cmd/core-command/res/configuration.toml#L93

Reply:
I think we just log an error in this case stating the issue and unable to publish error back to proper topic.

publishing to `edgex/command/response` will not help as the requestor would not be subscribed properly. 

Or maybe we should be publishing the response back to `edgex/command/response/<request-id>`

Still have the issue of how to send error responses when we don't get a requestId or error unmarshiling the request envelope. So best we can do is log the errors.

@cloudxxx8 , thoughts.

Reply:
After some consideration, I also think we should just write the error log for this kind of error.